src/skin/legacy/legacyskinparser.cpp

```diff
@@ -2426,26 +2430,30 @@ void LegacySkinParser::setupConnections(const QDomNode& node, WBaseWidget* pWidg
                 emitOption |= ControlParameterWidgetConnection::EMIT_DEFAULT;
             }
 
-            ControlParameterWidgetConnection* pConnection = new ControlParameterWidgetConnection(
-                    pWidget, control->getKey(), pTransformer,
-                    static_cast<ControlParameterWidgetConnection::DirectionOption>(directionOption),
-                    static_cast<ControlParameterWidgetConnection::EmitOption>(emitOption));
+            auto pConnection =
+                    std::make_unique<ControlParameterWidgetConnection>(pWidget,
+                            control->getKey(),
+                            std::move(pTransformer),
+                            static_cast<ControlParameterWidgetConnection::
+                                            DirectionOption>(directionOption),
+                            static_cast<ControlParameterWidgetConnection::
+                                            EmitOption>(emitOption));
 
             switch (state) {
             case Qt::NoButton:
-                pWidget->addConnection(pConnection);
                 if (directionOption & ControlParameterWidgetConnection::DIR_TO_WIDGET) {
-                    pLastLeftOrNoButtonConnection = pConnection;
+                    pLastLeftOrNoButtonConnection = pConnection.get();
                 }
+                pWidget->addConnection(std::move(pConnection));
                 break;
             case Qt::LeftButton:
-                pWidget->addLeftConnection(pConnection);
                 if (directionOption & ControlParameterWidgetConnection::DIR_TO_WIDGET) {
-                    pLastLeftOrNoButtonConnection = pConnection;
+                    pLastLeftOrNoButtonConnection = pConnection.get();
```

Comment:
it only stores it for the lifetime of the function. Its just an artifact that the code wants to use this at the end of the function again. I can easily move that part into the branch here so it acts on the unique_ptr directly and we don't need to retain it until later.

Reply:
I think I misunderstood. Instead you meant to move the `.get()` into the class, right? I tried to implement that in c9979c16bd455475e9e12a3bfba908f9bc276790, wdyt?

Reply:
That should work, however the name 
pWidget->setDisplayConnection() is confusing, because it also adds and keeps the connection. Only the last one becomes than the display connection. addConnectionAndUseForDisplay() or such. 
The alternative would be to integrate the whole logic in addConnection() or add a new call that calculated the display connection after all connections are added.

Reply:
The best comments are the one that can be deleted because the function name is significant enough. I have already proposed a function name and an alternative solution above. What do you think about them?

Reply:
I agree, though putting the entire documentation of the function into the name is just as bad. I just thought it was too long and the detail not significant enough. I can change it though if you insist.

Reply:
as per the alternative solution: Again, I currently don't have a good idea on how you would solve that issue which is why I asked you for a mockup:

> I'm not sure how making this part of addConnection would solve the issue exactly. Do you have a more concrete suggestion?

Reply:
The idea is to hide the display connection stuff in that function and just: 

```
void WBaseWidget::addConnection(
        std::unique_ptr<ControlParameterWidgetConnection> pConnection,
        ConnectionSide side) {
...
     int directionOption = pConnection->getDirectionOption();
     if (directionOption & ControlParameterWidgetConnection::DIR_TO_WIDGET) {
        m_pDisplayConnection = pConnection.get();
     }
}
```

Reply:
certainly interesting. I'll explore how feasible that is soon.

Reply:
Okay, After looking at this I'm unsure how good of an idea it is. 
My understanding is the following: The fact that last connection is automatically the `displayConnection` is (as far as I can tell by the comment) legacy behavior that is only there to provide stability for skins. `WBaseWidget` is however not intrinsically coupled to the skins thus moving this hack into the class mixes concerns. Moreover, the legacy behavior would now be backed into the class, so opting out of it would be impossible. I think that is a worse tradeoff than a bad or unwieldy long name. Wdyt?

Reply:
The comment about "Legacy" means "Keep it like that, to not introduce regressions."
The related review can be found here: https://github.com/mixxxdj/mixxx/pull/146#discussion-diff-8800549
@rryan suggested to explicit introduce a `<Display>` connection. But this has never been introduced. Since no one ask to have that in 10 years and we now have a dedicated hotcue button anyway, the legacy behavior is just fine. 

We may consider to adjust the comment accordingly. I don't mind if you go for a lengthy function name or for the widget move. The Widget move will match a similar call in the Hotcue button: https://github.com/mixxxdj/mixxx/blob/028e3ff4ee0ca464c7c47faa89a1c12a3e99b4e9/src/widget/whotcuebutton.cpp#L85

Reply:
so to clarify, you are okay with a better comment or name?

Reply:
The core issue is that the current name is misleading and should be changed. The issue is that becoming a display connection is only a side effect of the function call. Currently it looks like it is it's main purpose. If you don't find a perfect name, a different name AND a comment is fine.

Reply:
I changed the name and added a comment. Please review. If you don't like it, please make a concrete suggestion.

Reply:
Looks good. Thank you.

Reply:
Thank you. Waiting for more comments